nc/source/graphics/techniques/PhongShadingTechnique.h

```diff
@@ -3,6 +3,7 @@
 #include "graphics/techniques/Technique.h"
 #include "graphics/techniques/TechniqueType.h"
 #include "graphics/materials/MaterialProperties.h"
+#include "graphics/d3dresource/ConstantBufferResources.h"
```

Comment:
PixelConstantBuffer is forward declared in this file. Can this include be in the cpp? It may require the default destructor in cpp trick.